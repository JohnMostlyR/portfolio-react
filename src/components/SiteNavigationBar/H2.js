import styled from 'styled-components';

import {visuallyHidden} from '../../styles/mixins';

const H2 = styled.h2`
  ${visuallyHidden}
`;

export default H2;
