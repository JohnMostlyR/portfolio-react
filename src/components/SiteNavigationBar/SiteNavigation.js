import styled from 'styled-components';
import {rem} from 'polished';

import mq from '../../styles/templates/mediaQueries';

const SiteNavigation = styled.nav`
  width: 100%;
  padding: .35rem 0;
  position: ${props => props.isAtScreenTop ? 'fixed' : 'unset'};
  top: 0;
  background-color: #575756;

  ${mq.m`
    left: 100%;
    position: ${props => props.isAtScreenTop ? 'fixed' : 'absolute'};
    top: ${props => props.isAtScreenTop ? 0 : rem('77px')};
    width: unset;
    padding: 0 .35rem;
    transform: rotate(90deg);
    transform-origin: top left;
    background-color: transparent;
    border-bottom: none;  
  `}
`;

export default SiteNavigation;
