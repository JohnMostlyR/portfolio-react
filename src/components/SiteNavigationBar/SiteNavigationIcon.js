import styled from 'styled-components';
import FontAwesome from 'react-fontawesome';

import mq from '../../styles/templates/mediaQueries';

const SiteNavigationIcon = styled(FontAwesome)`
  display: none !important;

  ${mq.m`
    display: inherit !important;
    transform: rotate(-90deg);
    margin: 0;
  `}
`;

export default SiteNavigationIcon;
