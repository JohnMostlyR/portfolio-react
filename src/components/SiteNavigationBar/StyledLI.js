import styled from 'styled-components';

import mq from '../../styles/templates/mediaQueries';

const StyledLI = styled.li`
  vertical-align: middle;

  ${mq.m`margin-top: 2vw;`}

  &:not(:first-child) {
    ${mq.m`margin-left: 1vh;`}
  }
`;

export default StyledLI;
