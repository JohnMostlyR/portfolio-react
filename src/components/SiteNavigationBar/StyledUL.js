import styled from 'styled-components';
import mq from '../../styles/templates/mediaQueries';

const StyledUL = styled.ul`
  display: flex;
  justify-content: space-around;
  list-style: none;
  margin: 0;
  padding: 0;

  ${mq.m`justify-content: flex-start;`}
`;

export default StyledUL;
