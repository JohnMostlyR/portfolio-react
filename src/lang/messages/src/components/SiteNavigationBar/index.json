[
  {
    "id": "portfolio.nav.header",
    "defaultMessage": "Site navigation"
  }
]