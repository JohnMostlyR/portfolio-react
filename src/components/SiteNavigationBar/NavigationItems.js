import React from 'react';
import PropTypes from 'prop-types';
import {FormattedMessage} from 'react-intl';

import StyledUL from './StyledUL';
import StyledLI from './StyledLI';
import SiteNavLink from './SiteNavLink';
import SiteNavigationIcon from './SiteNavigationIcon';
import Description from './Description';

const NavigationItems = (props, context) => {
  return (
    <StyledUL>
      <StyledLI>
        <SiteNavLink to={'/'} data-isactive={context.router.route.location.pathname === '/'}>
          <SiteNavigationIcon name={'home'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.home'
                              defaultMessage='Intro'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/about'}
                     data-isactive={context.router.route.location.pathname === '/about'}>
          <SiteNavigationIcon name={'address-card'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.about'
                              defaultMessage='About'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/skills'}
                     data-isactive={context.router.route.location.pathname === '/skills'}>
          <SiteNavigationIcon name={'code'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.skills'
                              defaultMessage='Skills'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/projects'}
                     data-isactive={context.router.route.location.pathname === '/projects'}>
          <SiteNavigationIcon name={'list'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.projects'
                              defaultMessage='Projects'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
      <StyledLI>
        <SiteNavLink to={'/contact'}
                     data-isactive={context.router.route.location.pathname === '/contact'}>
          <SiteNavigationIcon name={'send'} fixedWidth/>
          <Description>
            <FormattedMessage id='portfolio.nav.contact'
                              defaultMessage='Contact'
            />
          </Description>
        </SiteNavLink>
      </StyledLI>
    </StyledUL>
  )
};

NavigationItems.contextTypes = {
  router: PropTypes.object,
};

export default NavigationItems;
