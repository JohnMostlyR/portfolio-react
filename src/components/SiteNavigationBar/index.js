import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {FormattedMessage} from 'react-intl';

import StyledUL from './StyledUL';
import StyledLI from './StyledLI';
import SiteNavLink from './SiteNavLink';
import SiteNavigationIcon from './SiteNavigationIcon';
import Description from './Description';
import SiteNavigation from './SiteNavigation';
import H2 from './H2';

import getElementTop from '../../utils/getElementTop';

class SiteNavigationBar extends Component {
  static propTypes = {
    setSiteNavIsFixedOffset: PropTypes.func.isRequired,
    siteNavIsAtScreenTop: PropTypes.bool.isRequired,
  };

  static contextTypes = {
    router: PropTypes.object,
  };

  componentDidMount() {
    const myTopPosition = getElementTop(this.siteNavigation) || 0;
    const myOffsetHeight = this.siteNavigation.offsetHeight;
    this.props.setSiteNavIsFixedOffset({
      siteNavigationTopPosition: myTopPosition,
      siteNavigationOffsetHeight: myOffsetHeight,
    });
  }

  componentWillUnmount() {
    this.props.setSiteNavIsFixedOffset({
      siteNavigationTopPosition: 0,
      siteNavigationOffsetHeight: 0,
    });
  }

  render() {
    const CURRENT_PATHNAME = this.context.router.route.location.pathname;

    return (
        <SiteNavigation isAtScreenTop={this.props.siteNavIsAtScreenTop}
                        innerRef={el => this.siteNavigation = el}>
          <H2><FormattedMessage id='portfolio.nav.header' defaultMessage='Site navigation'/></H2>
          <StyledUL>
            <StyledLI>
              <SiteNavLink to={'/'} data-isactive={CURRENT_PATHNAME === '/'}>
                <SiteNavigationIcon name={'home'} fixedWidth/>
                <Description>
                  <FormattedMessage id='portfolio.nav.home'
                                    defaultMessage='Intro'
                  />
                </Description>
              </SiteNavLink>
            </StyledLI>
            <StyledLI>
              <SiteNavLink to={'/about'}
                           data-isactive={CURRENT_PATHNAME === '/about'}>
                <SiteNavigationIcon name={'address-card'} fixedWidth/>
                <Description>
                  <FormattedMessage id='portfolio.nav.about'
                                    defaultMessage='About'
                  />
                </Description>
              </SiteNavLink>
            </StyledLI>
            <StyledLI>
              <SiteNavLink to={'/skills'}
                           data-isactive={CURRENT_PATHNAME === '/skills'}>
                <SiteNavigationIcon name={'code'} fixedWidth/>
                <Description>
                  <FormattedMessage id='portfolio.nav.skills'
                                    defaultMessage='Skills'
                  />
                </Description>
              </SiteNavLink>
            </StyledLI>
            <StyledLI>
              <SiteNavLink to={'/projects'}
                           data-isactive={CURRENT_PATHNAME === '/projects'}>
                <SiteNavigationIcon name={'list'} fixedWidth/>
                <Description>
                  <FormattedMessage id='portfolio.nav.projects'
                                    defaultMessage='Projects'
                  />
                </Description>
              </SiteNavLink>
            </StyledLI>
            <StyledLI>
              <SiteNavLink to={'/contact'}
                           data-isactive={CURRENT_PATHNAME === '/contact'}>
                <SiteNavigationIcon name={'send'} fixedWidth/>
                <Description>
                  <FormattedMessage id='portfolio.nav.contact'
                                    defaultMessage='Contact'
                  />
                </Description>
              </SiteNavLink>
            </StyledLI>
          </StyledUL>
        </SiteNavigation>
    );
  }
}

export default SiteNavigationBar;
