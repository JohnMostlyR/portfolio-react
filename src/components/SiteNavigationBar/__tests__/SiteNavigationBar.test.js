import React from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import createRouterContext from 'react-router-test-context';

import SiteNavigationBar from '../index';
import getElementTop from '../../../utils/getElementTop';

jest.mock('../../../utils/getElementTop');

describe('SiteNavigationBar', () => {
  const routerContext = createRouterContext();
  const setSiteNavIsFixedOffset = jest.fn();

  it('should render and match the snapshot', () => {
    const wrapper = mount(
        <SiteNavigationBar setSiteNavIsFixedOffset={setSiteNavIsFixedOffset}
                           siteNavIsAtScreenTop={false}/>,
        {context: routerContext},
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should get its own top position on the screen', () => {
    const wrapper = mount(
        <SiteNavigationBar setSiteNavIsFixedOffset={setSiteNavIsFixedOffset}
                           siteNavIsAtScreenTop={false}/>,
        {context: routerContext},
    );

    const invocationArgs = getElementTop.mock.calls[0];
    expect(invocationArgs[0].nodeType).toBe(1);
    wrapper.unmount();
  });

  it('should call setSiteNavIsFixedOffset', () => {
    const wrapper = mount(
        <SiteNavigationBar setSiteNavIsFixedOffset={setSiteNavIsFixedOffset}
                           siteNavIsAtScreenTop={false}/>,
        {context: routerContext},
    );

    const invocationArgs = setSiteNavIsFixedOffset.mock.calls[0];
    expect(invocationArgs[0]).toHaveProperty('siteNavigationTopPosition');
    expect(invocationArgs[0]).toHaveProperty('siteNavigationOffsetHeight');
    wrapper.unmount();
  });

  it('should act when top of the screen is reached', () => {
    const wrapper = mount(
        <SiteNavigationBar setSiteNavIsFixedOffset={setSiteNavIsFixedOffset}
                           siteNavIsAtScreenTop={true}/>,
        {context: routerContext},
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});