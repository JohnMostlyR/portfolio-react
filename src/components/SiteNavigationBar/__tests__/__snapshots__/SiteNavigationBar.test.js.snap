// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteNavigationBar should act when top of the screen is reached 1`] = `
<SiteNavigationBar
  setSiteNavIsFixedOffset={[Function]}
  siteNavIsAtScreenTop={true}
>
  <styled.nav
    innerRef={[Function]}
    isAtScreenTop={true}
  >
    <nav
      className="sc-EHOje jBrOTl"
    >
      <styled.h2>
        <h2
          className="sc-bZQynM pCdmS"
        >
          <FormattedMessage
            defaultMessage="Site navigation"
            id="portfolio.nav.header"
          >
            <IntlProvider
              defaultLocale="en"
              locale="en"
            >
              <FormattedMessage
                defaultMessage="Site navigation"
                id="portfolio.nav.header"
                values={Object {}}
              >
                <span>
                  Site navigation
                </span>
              </FormattedMessage>
            </IntlProvider>
          </FormattedMessage>
        </h2>
      </styled.h2>
      <styled.ul>
        <ul
          className="sc-bdVaJa hHvDrO"
        >
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={true}
                to="/"
              >
                <Link
                  className="sc-htpNat zgiCp"
                  data-isactive={true}
                  replace={false}
                  to="/"
                >
                  <a
                    className="sc-htpNat zgiCp"
                    data-isactive={true}
                    href="/"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="home"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="home"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-home fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Intro"
                          id="portfolio.nav.home"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Intro"
                              id="portfolio.nav.home"
                              values={Object {}}
                            >
                              <span>
                                Intro
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/about"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/about"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/about"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="address-card"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="address-card"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-address-card fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="About"
                          id="portfolio.nav.about"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="About"
                              id="portfolio.nav.about"
                              values={Object {}}
                            >
                              <span>
                                About
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/skills"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/skills"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/skills"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="code"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="code"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-code fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Skills"
                          id="portfolio.nav.skills"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Skills"
                              id="portfolio.nav.skills"
                              values={Object {}}
                            >
                              <span>
                                Skills
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/projects"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/projects"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/projects"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="list"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="list"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-list fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Projects"
                          id="portfolio.nav.projects"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Projects"
                              id="portfolio.nav.projects"
                              values={Object {}}
                            >
                              <span>
                                Projects
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/contact"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/contact"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/contact"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="send"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="send"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-send fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Contact"
                          id="portfolio.nav.contact"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Contact"
                              id="portfolio.nav.contact"
                              values={Object {}}
                            >
                              <span>
                                Contact
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
        </ul>
      </styled.ul>
    </nav>
  </styled.nav>
</SiteNavigationBar>
`;

exports[`SiteNavigationBar should render and match the snapshot 1`] = `
<SiteNavigationBar
  setSiteNavIsFixedOffset={[Function]}
  siteNavIsAtScreenTop={false}
>
  <styled.nav
    innerRef={[Function]}
    isAtScreenTop={false}
  >
    <nav
      className="sc-EHOje ejgrxl"
    >
      <styled.h2>
        <h2
          className="sc-bZQynM pCdmS"
        >
          <FormattedMessage
            defaultMessage="Site navigation"
            id="portfolio.nav.header"
          >
            <IntlProvider
              defaultLocale="en"
              locale="en"
            >
              <FormattedMessage
                defaultMessage="Site navigation"
                id="portfolio.nav.header"
                values={Object {}}
              >
                <span>
                  Site navigation
                </span>
              </FormattedMessage>
            </IntlProvider>
          </FormattedMessage>
        </h2>
      </styled.h2>
      <styled.ul>
        <ul
          className="sc-bdVaJa hHvDrO"
        >
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={true}
                to="/"
              >
                <Link
                  className="sc-htpNat zgiCp"
                  data-isactive={true}
                  replace={false}
                  to="/"
                >
                  <a
                    className="sc-htpNat zgiCp"
                    data-isactive={true}
                    href="/"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="home"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="home"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-home fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Intro"
                          id="portfolio.nav.home"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Intro"
                              id="portfolio.nav.home"
                              values={Object {}}
                            >
                              <span>
                                Intro
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/about"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/about"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/about"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="address-card"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="address-card"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-address-card fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="About"
                          id="portfolio.nav.about"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="About"
                              id="portfolio.nav.about"
                              values={Object {}}
                            >
                              <span>
                                About
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/skills"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/skills"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/skills"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="code"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="code"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-code fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Skills"
                          id="portfolio.nav.skills"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Skills"
                              id="portfolio.nav.skills"
                              values={Object {}}
                            >
                              <span>
                                Skills
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/projects"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/projects"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/projects"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="list"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="list"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-list fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Projects"
                          id="portfolio.nav.projects"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Projects"
                              id="portfolio.nav.projects"
                              values={Object {}}
                            >
                              <span>
                                Projects
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
          <styled.li>
            <li
              className="sc-bwzfXH kiBDtw"
            >
              <Styled(Link)
                data-isactive={false}
                to="/contact"
              >
                <Link
                  className="sc-htpNat ipIhTO"
                  data-isactive={false}
                  replace={false}
                  to="/contact"
                >
                  <a
                    className="sc-htpNat ipIhTO"
                    data-isactive={false}
                    href="/contact"
                    onClick={[Function]}
                  >
                    <Styled(FontAwesome)
                      fixedWidth={true}
                      name="send"
                    >
                      <FontAwesome
                        className="sc-bxivhb jYHDuC"
                        fixedWidth={true}
                        name="send"
                      >
                        <span
                          aria-hidden={true}
                          className="fa fa-send fa-fw sc-bxivhb jYHDuC"
                        />
                      </FontAwesome>
                    </Styled(FontAwesome)>
                    <styled.span>
                      <span
                        className="sc-ifAKCX kiQsQu"
                      >
                        <FormattedMessage
                          defaultMessage="Contact"
                          id="portfolio.nav.contact"
                        >
                          <IntlProvider
                            defaultLocale="en"
                            locale="en"
                          >
                            <FormattedMessage
                              defaultMessage="Contact"
                              id="portfolio.nav.contact"
                              values={Object {}}
                            >
                              <span>
                                Contact
                              </span>
                            </FormattedMessage>
                          </IntlProvider>
                        </FormattedMessage>
                      </span>
                    </styled.span>
                  </a>
                </Link>
              </Styled(Link)>
            </li>
          </styled.li>
        </ul>
      </styled.ul>
    </nav>
  </styled.nav>
</SiteNavigationBar>
`;
